LiverSegments/Logic/vtkLiverSegmentsLogic.cxx

```diff
@@ -0,0 +1,133 @@
+ /*===============================================================================
+
+   Project: LiverSegments
+   Module: vtkSegmentClassificationLogic.cxx
+
+   Copyright (c) 2019,  Oslo University Hospital
+
+   All rights reserved. This is propietary software. In no event shall the author
+   be liable for any claim or damages 
+
+   ===============================================================================*/
+
+#include "vtkLiverSegmentsLogic.h"
+
+#include <vtkMRMLLabelMapVolumeNode.h>
+#include <vtkMRMLSegmentationNode.h>
+#include <vtkMRMLModelNode.h>
+
+#include <vtkObjectFactory.h>
+#include <vtkImageData.h>
+#include <vtkImageIterator.h>
+#include <vtkKdTreePointLocator.h>
+#include <vtkPointSet.h>
+#include <vtkPolyData.h>
+#include <vtkPointData.h>
+#include <vtkStringArray.h>
+#include <vtkAppendPolyData.h>
+#include <vtkOrientedImageData.h>
+#include <vtkMatrix4x4.h>
+
+#include <iostream>
+
+//------------------------------------------------------------------------------
+vtkStandardNewMacro(vtkLiverSegmentsLogic);
+
+//------------------------------------------------------------------------------
+vtkLiverSegmentsLogic::vtkLiverSegmentsLogic()
+{
+  locator = vtkSmartPointer<vtkKdTreePointLocator>::New();
+}
+
+//------------------------------------------------------------------------------
+vtkLiverSegmentsLogic::~vtkLiverSegmentsLogic()
+{
+  
+}
+
+//------------------------------------------------------------------------------
+void vtkLiverSegmentsLogic::PrintSelf(ostream &os, vtkIndent indent)
+{
+  Superclass::PrintSelf(os, indent);
+}
+
+void vtkLiverSegmentsLogic::markSegmentWithID(vtkMRMLModelNode *segment, int segmentId)
+{
+    vtkSmartPointer<vtkIntArray> idArray = vtkSmartPointer<vtkIntArray>::New();
+    idArray->SetName("segmentId");
+    vtkSmartPointer<vtkPolyData> polydata = segment->GetPolyData();
```

Comment:
I would not use `vtkSmartPointer` in these situations, but simply `vtkPolyData`. It clutters more the code, it adds a little overhead to keep the smart pointer and it is not needed.

Sometimes I would also use just `auto`, which in this case resolves to `vtkPolyData*`